test/sqllogictest/transform/demand.slt

```diff
@@ -0,0 +1,103 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+mode cockroach
+
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM SET enable_with_mutually_recursive = true
+----
+COMPLETE 0
+
+statement ok
+CREATE TABLE t (a int)
+
+statement ok
+INSERT INTO t (a) VALUES (1), (2), (3);
+
+# Test that demand analysis does not result in a 1/0 error for column
```

Comment:
I think this will never result in a 1/0 error at the moment because you are not inserting `0` as a possible `a` value in `t` above.

Reply:
Hmm, yes, good catch! This was an old test, and I didn't actually check that it is correct before modifying it...